Small refactoring which makes possible to run latex check from node.js

dbarashev opened the pull request:
Hey Alex @dainiak , 
this refactoring will allow us to run your code on the server side. It should still work in the browser the same way as before.

dbarashev commented on js/main.js:
```diff
@@ -356,7 +356,7 @@ function initiate() {
             }
         }
 
-        if (typeof addWarningCustom === 'undefined') {
```
Why don't you just compare with `undefined`? `addWarningCustom == undefined` ?

dainiak commented:
> `addWarningCustom == undefined`

I suppose this should do:
`addWarningCustom === undefined`

dbarashev commented:
@dainiak @yanenkoa  with `==` actually because`null === undefined` is `false` while `null == undefined` is `true`. So, if someone passes `null` in `addWarningCustom` then this code blows up.

dainiak commented:
OK! Shall I merge it in the current state for now then?

dbarashev commented:
Wait a minute :)

On Mon, Feb 20, 2017 at 4:45 PM Alex Dainiak <notifications@github.com>
wrote:

> OK! Shall I merge it in the current state for now then?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dainiak/latexcheck/pull/1#issuecomment-281082866>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AB7zKuk4xhotlq_hkMx013ljEDM2ml6aks5reZj3gaJpZM4MBrQb>
> .
>

dbarashev commented:
I've replaced typeofs with comparison to undefined and fixed a bug. Must be
good to merge now.

On Mon, Feb 20, 2017 at 4:54 PM Dmitry Barashev <dbarashev@ganttproject.biz>
wrote:

Wait a minute :)

On Mon, Feb 20, 2017 at 4:45 PM Alex Dainiak <notifications@github.com>
wrote:

OK! Shall I merge it in the current state for now then?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dainiak/latexcheck/pull/1#issuecomment-281082866>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/AB7zKuk4xhotlq_hkMx013ljEDM2ml6aks5reZj3gaJpZM4MBrQb>
.
